vaadin/auth
# feat: add lumo based theme for Keycloak provider

## Description

Add a module to build a jar containing the lumo-based theme to be added to a Keycloak instance.

The theme contains a common CSS file with CSS properties and styles taken from Lumo. It would be nicer to have it injected as a npm dependency, so we wouldn't need to copy the styles, but it can be introduced as an enhancement in another ticket. 

### To test it locally:

- Build the module with `mvn package`
- Copy the `sso-kit-keycloak-theme-1.0-SNAPSHOT.jar` file from the `target` folder and paste it into the `providers` folder of your local Keycloak instance
- Start the Keycloak server
- Go to the Admin console and then "Realm settings > Themes" and under "Login theme", choose "sso-kit-keycloak-theme" and hit Save
- Go to the login page and check if the theme is correctly applied

### To customize your build using `sso-kit-keycloak-theme` as the parent theme:

- Under `themes`, create a folder structure like `your-theme-name/login` and then create these two files:
  - `theme.properties`
  - `resources/login/custom-styles.css`
- On `theme.properties`:
  ```properties
	parent=sso-kit-keycloak-theme

	styles=css/lumo-common.css css/lumo-styles.css css/custom-styles.css
  ```
- On `custom-styles.css`:
  ```css
	/* HERE YOU CAN CUSTOMIZE SOME CSS PROPERTIES LIKE THE PRIMARY COLORS, FOR INSTANCE */
	html {
      --lumo-primary-color: rgb(22, 138, 72);
      --lumo-primary-color-50pct: rgba(22, 138, 72, 0.704);
	}
  ```
- The new theme should be available on the same "Login theme" select mentioned previously

Fixes #18

## Type of change
